### PR TITLE
Add blind index search

### PR DESCRIPTION
**Pull request by giarc3**
TODO:
- [x] Depend on released ironoxide

### CODE REVIEW

File: src/lib.rs.in [hunk 1]
```diff
@@ -719,6 +719,33 @@ class Duration{
     pre_build_generate_equals_and_hashcode Duration;
 });
 
+foreign_class!(
+class EncryptedBlindIndexSalt {
```
Comment (coltfred) [inline src/lib.rs.in]: I think we need this type to have a constructor that takes the deks and data.

We should also have a test that does this to show how it works.

Comment (giarc3) [reply src/lib.rs.in]: Done (I think)

File: tests/src/test/scala/ironrust/FullIntegrationTest.scala [hunk 2]
```diff
@@ -570,6 +570,25 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
     }
   }
 
+  "Encrypted search" should {
+    "tokenize a query correctly" in {
```
Comment (coltfred) [inline tests/src/test/scala/ironrust/FullIntegrationTest.scala]: Might be nice to have a call to tokenizeData and verify that the result of tokenizeQuery is a subset of the result.

Comment (giarc3) [reply tests/src/test/scala/ironrust/FullIntegrationTest.scala]: Yep, I had meant to do that and forgot. On it now

Comment (giarc3) [reply tests/src/test/scala/ironrust/FullIntegrationTest.scala]: Done

File: tests/src/test/scala/ironrust/FullIntegrationTest.scala [hunk 3]
```diff
@@ -57,7 +57,7 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
       val regex = "(.*).java".r
       val rustSwigExclude = List("InternalPointerMarker", "JNIReachabilityFence")
       // any class that we can't implement equals and hashCode for must be in this list
-      val iclExclude = List("AssociationType", "IronOxide")
+      val iclExclude = List("AssociationType", "IronOxide", "BlindIndexSearch")
```
Comment (clintfred) [inline tests/src/test/scala/ironrust/FullIntegrationTest.scala]: We may, someday, need to implement equals and hashcode for BlindIndexSearch if we want to reasonable support having multiple search indexing in an application at a time as they might get stored in a hashmap. I think this is okay for now, however.

Comment (coltfred) [reply tests/src/test/scala/ironrust/FullIntegrationTest.scala]: @clintfred Because it holds a RNG, we could do it disregarding that, but it felt weird.

**Pull request merged**